### PR TITLE
[FIX] reorganize anat filename templates

### LINKED ISSUE #1418: [BUG] ambiguity in filename template in anat
### Describe your problem in detail.

See below.

The 2 last lines seem redundant because the part entity is already optional in the 2 first filename patterns.

```text
sub-<label>/
    [ses-<label>/]
        anat/
            sub-<label>[_ses-<label>][_task-<label>][_acq-<label>][_ce-<label>][_rec-<label>][_run-<index>][_part-<mag|phase|real|imag>]_<suffix>.json
            sub-<label>[_ses-<label>][_task-<label>][_acq-<label>][_ce-<label>][_rec-<label>][_run-<index>][_part-<mag|phase|real|imag>]_<suffix>.nii[.gz]
            sub-<label>[_ses-<label>][_task-<label>][_acq-<label>][_ce-<label>][_rec-<label>][_run-<index>]_<suffix>.json
            sub-<label>[_ses-<label>][_task-<label>][_acq-<label>][_ce-<label>][_rec-<label>][_run-<index>]_<suffix>.nii[.gz]
````

This seems to be due to fact that parametric anat files do not support the part entities where as non-param anat files do.

https://github.com/bids-standard/bids-specification/blob/master/src/schema/rules/files/raw/anat.yaml#L32

Both filename patterns get rendered but given that param and non-param have >6 suffixes it becomes impossible from the filename template alone to know which suffix supports the part entity.

### Describe what you expected.

Maybe only this? 

```text
sub-<label>/
    [ses-<label>/]
        anat/
            sub-<label>[_ses-<label>][_task-<label>][_acq-<label>][_ce-<label>][_rec-<label>][_run-<index>][_part-<mag|phase|real|imag>]_<suffix>.json
            sub-<label>[_ses-<label>][_task-<label>][_acq-<label>][_ce-<label>][_rec-<label>][_run-<index>][_part-<mag|phase|real|imag>]_<suffix>.nii[.gz]
```

Or better have `_<suffix>` be more specific to say which filename patterns goes with what suffix.

In any case given that we have a table in the `anat` section of all the non-parametric suffixes, we probably should have one with all the parametric ones.

### BIDS specification section

https://bids-specification.readthedocs.io/en/latest/modality-specific-files/magnetic-resonance-imaging-data.html#anatomy-imaging-data

### PR DESCRIPTION
**Pull request by Remi-Gau**
fixes #1418

- split filename templates macro for anat into:
  - parametric suffixes
  - defacemask
  - non parametric suffixes
- add list of file collection suffixes in qMRI appendix

### CODE REVIEW

**Comment by Remi-Gau** [comment]
@sappelhoff I labelled this as a fix but I am 100% sure this is accurate because the schema was correct but it was our rendering of it that created some ambiguity.

Hence not sure if this should actually go in the changelog.

**Comment by sappelhoff** [comment]
thanks! I think it should go into the changelog, because what users mostly see is what we render.

**Comment by Remi-Gau** [comment]
> thanks! I think it should go into the changelog, because what users mostly see is what we render.

good with me.
thanks!

File: src/modality-specific-files/magnetic-resonance-imaging-data.md [hunk 1]
```diff
@@ -165,16 +165,7 @@ whenever possible. See also
...
-This block generates a filename templates.
-The inputs for this macro can be found in the directory
-  src/schema/rules/files/raw
-and a guide for using macros can be found at
- https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
--->
-{{ MACROS___make_filename_template("raw", datatypes=["anat"]) }}
-
-Currently supported non-parametric structural MR images include:
+Currently supported non-parametric structural MR images include the following:
```
Comment (tsalo) [inline src/modality-specific-files/magnetic-resonance-imaging-data.md:168]: Maybe there could be a little intro to anatomical MRI here?

```suggestion
Anatomy MRI sequences measure static, structural features of the brain.
Each type of contrast has an associated suffix, resulting in many suffixes for this datatype.
For the sake of convenience, these suffixes are broadly divided into two groups: non-parametric and parametric.

Non-parametric suffixes have an arbitrary scale.
For example, T1w data are T1-weighted, but the values do not correspond to actual T1 value estimates.

Parametric suffixes, on the other hand, have a non-arbitrary scale.
A T1map file contains T1 value estimates, in seconds.

### Non-parametric structural MR images

Currently supported non-parametric structural MR images include the following:
```

File: src/modality-specific-files/magnetic-resonance-imaging-data.md [hunk 2]
```diff
@@ -207,6 +219,18 @@ entity corresponds to modality suffix,
...
+This block generates a filename templates.
+The inputs for this macro can be found in the directory
+  src/schema/rules/files/raw
+and a guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_filename_template("raw", datatypes=["anat"], suffixes=[
+         "defacemask",
+      ])
+}}
```
Comment (tsalo) [inline src/modality-specific-files/magnetic-resonance-imaging-data.md:232]: Can the defacemask stuff be moved to a new section _after_ the parametric suffixes?

File: src/modality-specific-files/magnetic-resonance-imaging-data.md [hunk 3]
```diff
@@ -207,6 +219,18 @@ entity corresponds to modality suffix,
...
+
 The OPTIONAL [`task-<label>`](../appendices/entities.md#task) entity can be used
 in order to allow tasks during structural MR acquisitions,
 for example pre-described motion paradigms such as nodding, to be described.
```
Comment (tsalo) [inline src/modality-specific-files/magnetic-resonance-imaging-data.md:236]: Maybe this could go after the parametric suffixes section.

**Review by tsalo — CHANGES_REQUESTED** [review]
I like the overall change, but I have some suggestions to make it flow more nicely.

File: src/modality-specific-files/magnetic-resonance-imaging-data.md (hunk 1 above)
Comment (tsalo) [reply src/modality-specific-files/magnetic-resonance-imaging-data.md:168]: This is a braindump. Definitely don't accept the suggestion as-is!

**Comment by Remi-Gau** [comment]
will have a closer look but it seems that I was being too conservative in my changes and that your suggestions really improve things.

File: src/modality-specific-files/magnetic-resonance-imaging-data.md (hunk 1 above)
Comment (Remi-Gau) [reply src/modality-specific-files/magnetic-resonance-imaging-data.md:168]: accepting this as is for now

will rework locally

File: src/modality-specific-files/magnetic-resonance-imaging-data.md (hunk 2 above)
Comment (Remi-Gau) [reply src/modality-specific-files/magnetic-resonance-imaging-data.md]: done

File: src/modality-specific-files/magnetic-resonance-imaging-data.md (hunk 3 above)
Comment (Remi-Gau) [reply src/modality-specific-files/magnetic-resonance-imaging-data.md]: done

**Review by tsalo — APPROVED** [review]
The changes look good to me. Thanks!

**Comment by Remi-Gau** [comment]
qmri appendix before
Screenshot from 2023-02-24 11-32-46


qmri appendix after
Screenshot from 2023-02-24 11-32-17

**Comment by Remi-Gau** [comment]
anat section before:

https://bids-specification.readthedocs.io/en/latest/modality-specific-files/magnetic-resonance-imaging-data.html#anatomy-imaging-data

Screenshot from 2023-02-24 11-35-02

anat section after:

https://bids-specification--1419.org.readthedocs.build/en/1419/modality-specific-files/magnetic-resonance-imaging-data.html#anatomy-imaging-data

Screenshot from 2023-02-24 11-31-18
Screenshot from 2023-02-24 11-31-33
Screenshot from 2023-02-24 11-31-40

**Comment by Remi-Gau** [comment]
@effigies 
added some screenshots and links to the relevant HTML to help reviewing.

let me know if you need something else.

**Comment by effigies** [comment]
Overall looks good, but I think I prefer the filename template to come before the suffix list.

**Comment by Remi-Gau** [comment]
> Overall looks good, but I think I prefer the filename template to come before the suffix list. Other modalities continue to follow this convention, so I think it would make more sense to keep that as-is for now. A global reordering could be proposed separately (though I would vote against it).

good point

**Comment by Remi-Gau** [comment]
> filename template to come before the suffix list.

@effigies 

This is done.

**Review by effigies — APPROVED** [review]

**Comment by Remi-Gau** [comment]
it's been 5 days so I will merge this

**Pull request merged**